src/Headless/Avalonia.Headless.NUnit/AvaloniaTestMethodCommand.cs

```diff
@@ -45,22 +50,75 @@ private AvaloniaTestMethodCommand(
 
     public static TestCommand ProcessCommand(HeadlessUnitTestSession session, TestCommand command)
     {
-        return ProcessCommand(session, command, new List<Action>(), new List<Action>());
+        return ProcessCommand(session, command, [], []);
     }
 
-    private static TestCommand ProcessCommand(HeadlessUnitTestSession session, TestCommand command, List<Action> before, List<Action> after)
+    private static TestCommand ProcessCommand(HeadlessUnitTestSession session, TestCommand command, List<Action> before, List<Action<TestExecutionContext>> after)
     {
         if (command is BeforeAndAfterTestCommand beforeAndAfterTestCommand)
         {
             if (s_beforeTest.GetValue(beforeAndAfterTestCommand) is Action<TestExecutionContext> beforeTest)
             {
-                Action<TestExecutionContext> beforeAction = c => before.Add(() => beforeTest(c));
-                s_beforeTest.SetValue(beforeAndAfterTestCommand, beforeAction);
+                var setUpTearDownInfo = beforeTest.Target?.GetType()
+                    .GetField("setUpTearDown", BindingFlags.Instance | BindingFlags.Public);
+                if (setUpTearDownInfo != null && setUpTearDownInfo.GetValue(beforeTest.Target) is SetUpTearDownItem setUpTearDown
+                    && s_setUpMethods.GetValue(setUpTearDown) is List<IMethodInfo> setUpMethods)
+                {
+                    Action<TestExecutionContext> beforeAction = c =>
+                    {
+                        before.AddRange(setUpMethods.Select<IMethodInfo, Action>(m => async void () =>
```

Comment:
`async void` won't work correctly: only _part_ of the async methods (until the first `await`) will run as expected, with the continuation running out of order (possibly even after the test), or not at all.

The `before` and `after` list should probably have items of type `Func<Task>`, with awaiting taking place in `ExecuteTestMethod`.